qa/standalone/mon/osd-pool-df.sh

```diff
@@ -73,4 +73,111 @@ function TEST_ceph_df() {
     teardown $dir || return 1
 }
 
+function TEST_stretched_cluster_ceph_df() {
+    local dir=$1
+    local OSDS=12
+    local nodes=6
+    setup $dir || return 1
+
+    run_mon $dir a || return 1
+    run_mgr $dir x || return 1
+    export CEPH_ARGS
+
+    for osd in $(seq 0 $(expr $OSDS - 1))
+    do
+      run_osd $dir $osd || return 1
+    done
+
+    for dc in DC1 DC2
+    do
+      ceph osd crush add-bucket $dc datacenter
+      ceph osd crush move $dc root=default
+    done
+
+    for node in $(seq 0 $(expr $nodes - 1))
+    do
+      ceph osd crush add-bucket osd-$node host
+    done
+
+    for p in $(seq  0 2)
+    do
+      ceph osd crush move osd-$p datacenter=DC1
+    done
+
+    for q in $(seq  3 5)
+    do
+      ceph osd crush move osd-$q datacenter=DC2
+    done
+
+    ceph osd crush move osd.0 host=osd-0
+    ceph osd crush move osd.3 host=osd-0
+    ceph osd crush move osd.1 host=osd-1
+    ceph osd crush move osd.2 host=osd-1
+    ceph osd crush move osd.7 host=osd-2
+    ceph osd crush move osd.11 host=osd-2
+    ceph osd crush move osd.4 host=osd-3
+    ceph osd crush move osd.8 host=osd-3
+    ceph osd crush move osd.5 host=osd-4
+    ceph osd crush move osd.9 host=osd-4
+    ceph osd crush move osd.6 host=osd-5
+    ceph osd crush move osd.10 host=osd-5
+
+    hostname=$(hostname -s)
+    ceph osd crush remove $hostname
+
+    ceph osd getcrushmap > $dir/crushmap || return 1
+    crushtool --decompile $dir/crushmap > $dir/crushmap.txt || return 1
+    sed 's/^# end crush map$//' $dir/crushmap.txt > $dir/crushmap_modified.txt || return 1
+    cat >> $dir/crushmap_modified.txt << EOF
+rule stretch_rule {
+        id 1
+        type replicated
+        min_size 1
+        max_size 10
+        step take DC1
+        step chooseleaf firstn 2 type host
+        step emit
+        step take DC2
+        step chooseleaf firstn 2 type host
+        step emit
+}
+
+rule stretch_replicated_rule {
+        id 2
+        type replicated
+        min_size 1
+        max_size 10
+        step take default
+        step choose firstn 0 type datacenter
+        step chooseleaf firstn 2 type host
+        step emit
+}
+
+# end crush map
+EOF
+
+    crushtool --compile $dir/crushmap_modified.txt -o $dir/crushmap.bin || return 1
+    ceph osd setcrushmap -i $dir/crushmap.bin  || return 1
+
+    ceph osd crush rule ls
+
+    local stretched_poolname=stretched_rbdpool
+    local stretched_rep_poolname=stretched_replicated_rbdpool
+    ceph osd pool create $stretched_poolname 32 32 stretch_rule
+    ceph osd pool create $stretched_rep_poolname 32 32 stretch_replicated_rule
+    ceph osd pool set $stretched_poolname size 4
+    ceph osd pool set $stretched_rep_poolname size 4
+    wait_for_clean || return 1
+
+    local global_avail=`ceph df -f json | jq '.stats.total_avail_bytes'`
```

Comment:
Do we need this variable? I don't see it used anywhere in this test function.

Reply:
Yes, I suppose to remove this variable but missed it. Let me remove it.